### ISSUE TITLE
[BUG] [v1.32.1] Browser closing after test is failed

### ISSUE THREAD

**Issue by hardeepCDX** [microsoft/playwright#22076]
Given I run the test case through Playwright Test for VSCode extension
When The test case failed 
Then The browser is closed 

Expected Result: The browser shouldn't closed once the test case is failed

PS - I have noticed the browser is also getting closed once the test execution is finish which wasn't happening in the previous versions.

**Comment by yury-s** [comment]
This sounds like the same problem as in https://github.com/microsoft/playwright/issues/21993, please share your playwright config. Can you also try to change 'trace' to 'off' and see if the problem persists?

**Comment by hardeepCDX** [comment]
> Same issue with 'trace' to off.

> Playwright config:

import type { PlaywrightTestConfig } from '@playwright/test';
import { devices } from '@playwright/test';
import path from 'path';

/**
 * Read environment variables from file.
 * https://github.com/motdotla/dotenv
 * This code reads an environment variable ENV and uses it to determine which file to use to load environment 
 variables from using the dotenv package. If ENV is not set, it defaults to hard coded value.
 */
const env = process.env.ENV || 'uat';
require('dotenv').config({ path: path.resolve(__dirname, `.env.${env}`) });

/**
 * See https://playwright.dev/docs/test-configuration.
 */
const config: PlaywrightTestConfig = {
  testDir: './tests',
  /* Maximum time one test can run for. */
  timeout: 240000,
  expect: {
    /**
     * Maximum time expect() should wait for the condition to be met.
     * For example in `await expect(locator).toHaveText();`
     */
    timeout: 5000,
  },
  /* Run tests in files in parallel */
  fullyParallel: false,
  /* Fail the build on CI if you accidentally left test.only in the source code. */
  forbidOnly: !!process.env.CI,
  /* Retry on CI only */
  retries: process.env.CI ? 2 : 0,
  /* Opt out of parallel tests on CI. */
  workers: process.env.CI ? 1 : 1,
  /* Reporter to use. See https://playwright.dev/docs/test-reporters */
  reporter: 'html',
  /* Shared settings for all the projects below. See https://playwright.dev/docs/api/class-testoptions. */
  use: {
    /* Maximum time each action such as `click()` can take. Defaults to 0 (no limit). */
    actionTimeout: 30000,
    /* Base URL to use in actions like `await page.goto('/')`. */
    baseURL: process.env.TRADER_BASE_URL,
    /* Take screenshot when the test is failed. See https://playwright.dev/docs/screenshots */
    screenshot: {
      mode: 'only-on-failure',
      fullPage: false,
    },
    /* Collect trace when the test is failed. See https://playwright.dev/docs/trace-viewer */
    trace: 'off',
    /* Whether to ignore HTTPS errors when sending network requests. Defaults to false. */
    ignoreHTTPSErrors: true,
    /* Options used to launch the browser */
    launchOptions: {
      /* Use slowMo to slow down execution */
      slowMo: 500,
      /* Pass the --disable-web-security flag */
      args: ['--disable-web-security'],
    },
    /* Set the storageState based on CI environment, empty or 'state.json'. */
    storageState: process.env.CI ? '' : 'state.json',
  },

  /* Configure projects for major browsers */
  projects: [
    {
      name: 'chromium',
      use: {
        ...devices['Desktop Chrome'],
      },
    },

    {
      name: 'firefox',
      use: {
        ...devices['Desktop Firefox'],
      },
    },

    {
      name: 'webkit',
      use: {
        ...devices['Desktop Safari'],
      },
    },

    /* Test against mobile viewports. */
    {
      name: 'mobile-chrome',
      use: {
        ...devices['Galaxy S8'],
      },
    },
    {
      name: 'mobile-safari',
      use: {
        ...devices['iPhone 13'],
      },
    },

    /* Test against branded browsers. */
    {
      name: 'edge',
      use: {
        channel: 'msedge',
      },
    },
    {
      name: 'chrome',
      use: {
        channel: 'chrome',
      },
    },
  ],
};

export default config;

**Comment by hardeepCDX** [comment]
I have found this issue in v1.31.2 as well.

**Comment by yury-s** [comment]
I cannot reproduce it with 1.31.2 and a simple test. Please share full project that we can run locally to reproduce the problem.

**Comment by hardeepCDX** [comment]
Sorry, I am unable to disclose the project. However, I can provide you with instructions on how to replicate it:

1. Generate a test suite.
2. Introduce an error into the second test.
3. Execute the test suite.
4. As soon as the second test malfunctions, the browser will be shut down and restarted.
![image](https://user-images.githubusercontent.com/117135138/228701757-7fd52546-e164-4ff7-a946-1b7d5919f970.png)

**Comment by yury-s** [comment]
We cannot take any action on this report without a repro. Please file a new issue if you manage to extract a reproducible example that you can share and link it to this one.
